drivers/staging/media/rpivid/rpivid_h265.c

```diff
@@ -579,78 +599,26 @@ static void write_bitstream(struct rpivid_dec_env *const de,
 
 //////////////////////////////////////////////////////////////////////////////
 
-static void write_slice(struct rpivid_dec_env *const de,
-			const struct rpivid_dec_state *const s,
-			const unsigned int slice_w,
-			const unsigned int slice_h)
-{
-	u32 u32 = (s->sh->slice_type << 12) +
-		  (((s->sh->flags &
-		     V4L2_HEVC_SLICE_PARAMS_FLAG_SLICE_SAO_LUMA) != 0)
-		   << 14) +
-		  (((s->sh->flags &
-		     V4L2_HEVC_SLICE_PARAMS_FLAG_SLICE_SAO_CHROMA) != 0)
-		   << 15) +
-		  (slice_w << 17) + (slice_h << 24);
-
-	u32 |= (s->max_num_merge_cand << 0) + (s->nb_refs[L0] << 4) +
-	       (s->nb_refs[L1] << 8);
-
-	if (s->sh->slice_type == HEVC_SLICE_B)
-		u32 |= ((s->sh->flags &
-			 V4L2_HEVC_SLICE_PARAMS_FLAG_MVD_L1_ZERO) != 0)
-		       << 16;
-	p1_apb_write(de, RPI_SLICE, u32);
-}
-
-//////////////////////////////////////////////////////////////////////////////
-// Tiles mode
-
-static void new_entry_point(struct rpivid_dec_env *const de,
-			    const struct rpivid_dec_state *const s,
-			    const int do_bte,
-			    const int reset_qp_y, const int ctb_addr_ts)
+/*
+ * The slice constant part of the slice register - width and height need to
+ * be ORed in later as they are per-tile / WPP-row
+ */
+static u32 slice_reg_const(const struct rpivid_dec_state *const s)
 {
-	int ctb_col = s->ctb_addr_ts_to_rs[ctb_addr_ts] %
-							de->pic_width_in_ctbs_y;
-	int ctb_row = s->ctb_addr_ts_to_rs[ctb_addr_ts] /
-							de->pic_width_in_ctbs_y;
-
-	int tile_x = ctb_to_tile(ctb_col, s->col_bd, s->num_tile_columns);
-	int tile_y = ctb_to_tile(ctb_row, s->row_bd, s->num_tile_rows);
-
-	int endx = s->col_bd[tile_x + 1] - 1;
-	int endy = s->row_bd[tile_y + 1] - 1;
-
-	u8 slice_w = ctb_to_slice_w_h(ctb_col, 1 << s->log2_ctb_size,
-				      s->sps.pic_width_in_luma_samples,
-				      s->col_bd, s->num_tile_columns);
-	u8 slice_h = ctb_to_slice_w_h(ctb_row, 1 << s->log2_ctb_size,
-				      s->sps.pic_height_in_luma_samples,
-				      s->row_bd, s->num_tile_rows);
-
-	p1_apb_write(de, RPI_TILESTART,
-		     s->col_bd[tile_x] + (s->row_bd[tile_y] << 16));
-	p1_apb_write(de, RPI_TILEEND, endx + (endy << 16));
-
-	if (do_bte)
-		p1_apb_write(de, RPI_BEGINTILEEND, endx + (endy << 16));
-
-	write_slice(de, s, slice_w, slice_h);
-
-	if (reset_qp_y) {
-		unsigned int sps_qp_bd_offset =
-			6 * s->sps.bit_depth_luma_minus8;
-
-		p1_apb_write(de, RPI_QP, sps_qp_bd_offset + s->slice_qp);
-	}
-
-	p1_apb_write(de, RPI_MODE,
-		     (0xFFFF << 0) + (0x0 << 16) +
-			     ((tile_x == s->num_tile_columns - 1) << 17) +
-			     ((tile_y == s->num_tile_rows - 1) << 18));
-
-	p1_apb_write(de, RPI_CONTROL, (ctb_col << 0) + (ctb_row << 16));
+	u32 x = (s->max_num_merge_cand << 0) |
+		(s->nb_refs[L0] << 4) |
+		(s->nb_refs[L1] << 8) |
+		(s->sh->slice_type << 12);
+
+	if (s->sh->flags & V4L2_HEVC_SLICE_PARAMS_FLAG_SLICE_SAO_LUMA)
+		x |= BIT(14);
```

Comment:
Do we not have defines for the bit usage in these registers?

Reply:
No we don't - I got as far as creating defines for the registers but not individual bits.  There are a lot of bits.